# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/quota/Client.cpp
# Commit: c8a45415dc5b
# Full Hash: c8a45415dc5b0b33b7e9d4b9cd8114ac09a05218
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2021-05-28 21:48:03
# Description:
#   Bug 1706998: Replace QuotaManager::GetRef().MaybeRecordShutdownStep by static QuotaManager::(Safe)MaybeRecordQuotaClientShutdownStep; r=dom-storage-reviewers,janv
#   
#   There are some few unsafe uses of MaybeRecordShutdownStep where the QuotaManager singleton may be  not (yet or anymore) alive.
#   In order to not add many unnecessary null checks, we drop GetRef and MaybeRecordShutdownStep in favor of
#   QuotaManager::MaybeRecordQuotaClientShutdownStep
# ==============================================================================

diff -r cc21fb9be25a -r c8a45415dc5b dom/quota/Client.cpp
--- a/dom/quota/Client.cpp	Fri May 28 05:49:26 2021 +0000
+++ b/dom/quota/Client.cpp	Fri May 28 06:46:04 2021 +0000
@@ -233,7 +233,7 @@
 bool Client::InitiateShutdownWorkThreads() {
   AssertIsOnBackgroundThread();
 
-  QuotaManager::GetRef().MaybeRecordShutdownStep(GetType(), "starting"_ns);
+  QuotaManager::MaybeRecordQuotaClientShutdownStep(GetType(), "starting"_ns);
 
   InitiateShutdown();
 
@@ -241,7 +241,7 @@
 }
 
 void Client::FinalizeShutdownWorkThreads() {
-  QuotaManager::GetRef().MaybeRecordShutdownStep(GetType(), "completed"_ns);
+  QuotaManager::MaybeRecordQuotaClientShutdownStep(GetType(), "completed"_ns);
 
   FinalizeShutdown();
 }